### PR TITLE
NativeAOT: remove some headers that are not needed

### PR DESCRIPTION
**Pull request by akoeplinger**
Just a small cleanup.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/unix/configure.cmake [hunk 1]
```diff
@@ -15,9 +15,6 @@ endif()
 
 list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_FILE_OFFSET_BITS=64)
 
-check_include_files(sys/vmparam.h HAVE_SYS_VMPARAM_H)
-check_include_files(mach/vm_types.h HAVE_MACH_VM_TYPES_H)
-check_include_files(mach/vm_param.h HAVE_MACH_VM_PARAM_H)
```
Comment (am11) [inline src/coreclr/nativeaot/Runtime/unix/configure.cmake:20]: `HAVE_PTHREAD_ATTR_GET_NP`, `HAVE_SCHED_GETCPU`, `HAVE_CLOCK_MONOTONIC`, `HAVE_CLOCK_MONOTONIC_COARSE` and `HAVE_UCONTEXT_T` are also unused.

Comment (akoeplinger) [reply src/coreclr/nativeaot/Runtime/unix/configure.cmake:20]: no, those are both used from what I can see

Comment (am11) [reply src/coreclr/nativeaot/Runtime/unix/configure.cmake:20]: HAVE_UCONTEXT_T is in config.h.in, which can be deleted.

Yes, HAVE_PTHREAD_ATTR_GET_NP is in use.

Comment (akoeplinger) [reply src/coreclr/nativeaot/Runtime/unix/configure.cmake:20]: Thanks

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**